scan-payment/src/main/java/com/getbouncer/scan/payment/Image.kt

```diff
@@ -86,159 +94,182 @@ fun hasOpenGl31(context: Context): Boolean {
 /**
  * Fragments the image into multiple segments and places them in new segments.
  */
-fun Bitmap.fragment(
-    fromSegments: Array<Rect>,
-    toSegments: Array<Rect>,
-    toSize: Size
+fun Bitmap.rearrangeBySegments(
+    fromSegments: Map<Rect, Rect>,
+    futureImageSize: Size
```

Comment:
let's remove this since we can calculate it based on the `toSegments`

Reply:
The reason this is in there is because I have to create the bitmap + canvas at the beginning of the method. Though We could technically iterate through all the segments and find the size that way, that means we would have to iterate through all the segments twice, once to extrapolate the size to make the canvas, and then once more to actually draw onto the canvas. I personally feel that passing in a size will have a negligible performance impact in comparison to extrapolating the size every time.